core/internal/gqlmock/matchers.go

```diff
@@ -63,8 +64,14 @@ func (m *queryVariablesMatcher) Matches(x any) bool {
 		return false
 	}
 
+	// Match values at nested paths, if needed
 	for _, variable := range m.varMatchers {
-		if !variable.Value.Matches(varmap[variable.Name]) {
+		var curr interface{}
+		curr = varmap
+		for _, key := range strings.Split(variable.Name, ".") {
+			curr = curr.(map[string]any)[key]
+		}
+		if !variable.Value.Matches(curr) {
 			return false
 		}
 	}
```

Comment:
context: variable matching logic in `gqlmock` adapted to allow assertions on nested fields, e.g. `input.entityName` rather than just top-level fields like `entityName`